packages/legend-query/src/components/QueryBuilderFunctionPanel.tsx

```diff
@@ -0,0 +1,653 @@
+      props;
+    const queryFunctionsState = queryBuilderState.queryFunctionsState;
+    const treeData = queryFunctionsState.nonNullableTreeData;
+    const onNodeSelect = (node: QueryBuilderPackageElementTreeNodeData): void =>
+      queryFunctionsState.onTreeNodeSelect(queryBuilderState, node, treeData);
+
+    const getChildNodes = (
+      node: QueryBuilderPackageElementTreeNodeData,
+    ): QueryBuilderPackageElementTreeNodeData[] =>
+      getTreeChildNodes(queryBuilderState, node, treeData);
+
+    const dependencyTreeData = queryFunctionsState.getTreeData(
+      ROOT_PACKAGE_NAME.PROJECT_DEPENDENCY_ROOT,
+    );
+    const onDependencyTreeSelect = (
+      node: QueryBuilderPackageElementTreeNodeData,
+    ): void => {
+      if (dependencyTreeData) {
+        queryFunctionsState.onTreeNodeSelect(
+          queryBuilderState,
+          node,
+          dependencyTreeData,
+          ROOT_PACKAGE_NAME.PROJECT_DEPENDENCY_ROOT,
+        );
+      }
+    };
+
+    const getDependencyTreeChildNodes = (
+      node: QueryBuilderPackageElementTreeNodeData,
+    ): QueryBuilderPackageElementTreeNodeData[] => {
+      if (dependencyTreeData) {
+        return getTreeChildNodes(
+          queryBuilderState,
+          node,
+          dependencyTreeData,
+          ROOT_PACKAGE_NAME.PROJECT_DEPENDENCY_ROOT,
+        );
+      }
+      return [];
+    };
+
+    return (
+      <>
+        {rootPackageName === ROOT_PACKAGE_NAME.MAIN && (
+          <TreeView
+            components={{
+              TreeNodeContainer: QueryBuilderFunctionExplorerTreeNodeContainer,
+              TreeNodeView: QueryBuilderFunctionExplorerTreeNodeView,
+            }}
+            className="query-builder-function-explorer-tree__root"
+            treeData={treeData}
+            onNodeSelect={onNodeSelect}
+            getChildNodes={getChildNodes}
+            innerProps={{
+              queryBuilderState,
+              fromDependency: false,
+            }}
+          />
+        )}
+        {rootPackageName === ROOT_PACKAGE_NAME.PROJECT_DEPENDENCY_ROOT &&
+          dependencyTreeData &&
+          showDependencyFuncions && (
+            <>
+              <div className="query-builder__functions__separator" />
+              <TreeView
+                components={{
+                  TreeNodeContainer:
+                    QueryBuilderFunctionExplorerTreeNodeContainer,
+                  TreeNodeView: QueryBuilderFunctionExplorerTreeNodeView,
+                }}
+                className="query-builder-function-explorer-tree__root"
+                treeData={dependencyTreeData}
+                onNodeSelect={onDependencyTreeSelect}
+                getChildNodes={getDependencyTreeChildNodes}
+                innerProps={{
+                  queryBuilderState,
+                  fromDependency: true,
+                }}
+              />
+            </>
+          )}
+      </>
+    );
+  },
+);
+
+export const QueryBuilderFunctionPanel = observer(
```

Comment:
`QueryBuilderFunctionsExplorerPanel`, let's also rename this file and its related states in `QueryBuilderState`